src/components/tag/tag.js

```diff
@@ -1,51 +1,96 @@
 import React from 'react';
 import { Link } from 'react-router-dom';
 import PropTypes from 'prop-types';
-import classnames from 'classnames';
 import { css } from '@emotion/core';
 import vars from '../../../materials/custom-properties';
 import Constraints from '../constraints';
 import AccessibleButton from '../buttons/accessible-button';
 import Text from '../typography/text';
 import { CloseBoldIcon } from '../icons';
-import styles from './tag.mod.css';
 
 const getTextDetailColor = isDisabled => {
   if (isDisabled) return vars.colorGray60;
   return vars.colorBlack;
 };
 
-const getWrapperTypeClassName = type =>
-  type === 'warning' ? styles.wrapperTypeWarning : styles.wrapperTypeNormal;
-const getContentWrapperTypeClassName = type =>
+const getContentWrapperStyles = props => css`
+  display: flex;
+  box-sizing: border-box;
+  align-items: center;
+  min-height: ${vars.sizeHeightTag};
+  border-radius: ${vars.borderRadiusTag};
+  padding: 5px ${vars.spacing8};
+  cursor: default;
+  font-family: inherit;
+  white-space: normal;
+  text-align: left;
+  min-width: 0;
+  overflow-wrap: break-word;
+  hyphens: auto;
+  border-style: solid;
+  border-width: 1px;
+  border-color: ${props.type === 'warning'
+    ? vars.borderColorTagWarning
+    : vars.borderColorTagPristine};
+
+  /* fixing things for IE11 ... */
+  width: 100%;
+`;
+
+const getWrapperBackgroundColor = type =>
   type === 'warning'
-    ? styles.contentWrapperWarning
-    : styles.contentWrapperNormal;
-const getClickableContentWrapperTypeClassName = ({ type, isRemovable }) =>
+    ? vars.backgroundColorTagWarning
+    : vars.backgroundColorTagPristine;
+
+const getClickableContentWrapperStyles = ({ type, isRemoveable }) =>
   type === 'warning'
-    ? classnames(styles.clickableContentWrapperWarning, {
-        [styles.clickableRemovableContentWrapperWarning]: isRemovable,
-      })
-    : classnames(styles.clickableContentWrapperNormal, {
-        [styles.clickableRemovableContentWrapperNormal]: isRemovable,
-      });
+    ? [
```

Comment:
Just wondering if we could even do something like:

```js
const getClickableContentWrapperStyles = ({ type, isRemoveable }) => [
  type === 'warning' && css`
    &:hover {
      border-color: ${vars.borderColorTagWarningHover};
      border-right: 1px solid ${vars.borderColorTagPristine};
    }
  `,
  type === 'warning' && isRemoveable && css`
    border-right: 1px solid var(--border-color-tag-warning);
  `

  type =!== 'warning' && css`
    &:hover {
      border-color: ${vars.borderColorTagFocus};
    }
  `
  type =!== 'warning' && isRemoveable && css`
    &:hover {
      border-right: 1px solid ${vars.borderColorTagPristine};
    }
  `
].filter(Boolean)
```

Not sure maybe that's easier to follow. Just what I thought when reading this.

Reply:
I find the whole component hard to follow, I would rather refactor it not to be duplicated with the Link and the Regular, and I find this even more confusing

Reply:
Sure, it's subjective in the end just wanted to make a suggestion.